_static/adjust_logo_size.css

```diff
@@ -0,0 +1,5 @@
+/* Make the site logo smaller */
+
+.sidebar-logo {
+    width: 50%;
+}
```

Comment:
Wouldn't it be better to just resize the logo?
By resizing it, the page will load faster because it will have to download a smaller image and one less CSS file.

Reply:
It will also render faster and avoid layout shifts if we set explicit width and height in pixels, because the browser knows how much space the image needs before downloading it.

Reply:
The current image is 512x512 pixels. With `width: 50%` I get computed sizes between 114px and 103px at different window widths. Which should we pick?

A

Reply:
I'd suggest 128 px by 128 px and then make max-width so it is full size at regular widths, so its a clean 4x downsample which both looks much crisper and compresses much more efficiently.

Reply:
@hugovk you said

> It will also render faster and avoid layout shifts if we set explicit width and height in pixels

This means we still need the CSS file, correct?

A

Reply:
FWIW, even though `width` and `height` can be set as [HTML attributes of the `img` tag](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/img#attributes) and even though the `.. image:` directive should support the `:width:` and `:height:` options, this image is added through the [`html_logo` conf option](https://www.sphinx-doc.org/en/master/usage/configuration.html#confval-html_logo) and that doesn't seem to offer any further options or control.